os/include/tinyara/sched.h

```diff
@@ -221,6 +221,12 @@ typedef enum tstate_e tstate_t;
 #define FIRST_BLOCKED_STATE      TSTATE_TASK_INACTIVE
 #define LAST_BLOCKED_STATE       (NUM_TASK_STATES-1)
 
+/* This gives number of alive tasks at any point of time in the system.
+ * If the system is already running CONFIG_MAX_TASKS, Creating new
+ * task is not supported.
+ */
+extern volatile uint8_t g_alive_taskcount;
```

Comment:
This global variable is used only in kernel. Let's move it to kernel inside. Refer other global variable like g_inactivetasks.